sw/device/lib/dif/dif_pinmux.h

```diff
@@ -0,0 +1,502 @@
+ */
+typedef enum dif_pinmux_lock_type {
+  /**
+   * Lock Padring MIO input select for a given peripheral input. This means that
+   * the peripheral input is connection can no longer be changed.
+   */
+  kDifPinmuxLockTypeInsel = 0,
+  /**
+   * Lock peripheral output select for a given Padring MIO output. This means
+   * that the Padring MIO output connection can no longer be changed.
+   */
+  kDifPinmuxLockTypeOutsel,
+  /**
+   * Lock the sleep mode configuration for a given Padring MIO pin.
+   */
+  kDifPinmuxLockTypeMioSleep,
+  /**
+   * Lock the sleep mode configuration for a given Padring DIO pin.
+   */
+  kDifPinmuxLockTypeDioSleep,
+  /**
+   * Lock physical characteristics configuration for a given Padring MIO pin.
+   */
+  kDifPinmuxLockTypeMioPinAttr,
+  /**
+   * Lock physical characteristics configuration for a given Padring DIO pin.
+   */
+  kDifPinmuxLockTypeDioPinAttr,
+} dif_pinmux_lock_type_t;
+
+/**
+ * Pinmux generic index.
+ *
+ * The meaning of this index changes between APIs, and must be documented in
+ * the corresponding function description.
+ */
+typedef uint32_t dif_pinmux_index_t;
+
+/**
+ * Multiplexed Padring input connection select for a peripheral input.
+ */
+typedef uint32_t dif_pinmux_padring_insel_t;
+
+/**
+ * Multiplexed peripheral output connection select for a Padring output.
+ */
+typedef uint32_t dif_pinmux_peripheral_outsel_t;
+
+/**
+ * Pinmux Padring pin attributes.
+ *
+ * Some attributes have defined meanings, others are implementation defined. Not
+ * all pins will support all attributes. The Pinmux DIF does not check for
+ * unsupported attributes. To check to see if an attribute is supported after
+ * enabling it (via `dif_pinmux_padring_attr_enable` or
+ * `dif_pinmux_padring_attr_write`) read back the attribute field (via
+ * `dif_pinmux_padring_attr_get`). If the attribute isn't supported the
+ * attribute will still be disabled (relevant bit unset in the attribute field).
+ */
+typedef enum dif_pinmux_padring_pin_attr {
+  kDifPinmuxPadringPinAttrNone = 0,
+  kDifPinmuxPadringPinAttrIOInvert = 1 << 0,
+  kDifPinmuxPadringPinAttrOpenDrain = 1 << 1,
+  kDifPinmuxPadringPinAttrPullDown = 1 << 2,
+  kDifPinmuxPadringPinAttrPullUp = 1 << 3,
+  kDifPinmuxPadringPinAttrKeeper = 1 << 4,
+  kDifPinmuxPadringPinAttrWeakestDrive = 1 << 5,
+  kDifPinmuxPadringPinAttrWeakDrive = 1 << 6,
+  kDifPinmuxPadringPinAttrStrongDrive = 1 << 7,
+  kDifPinmuxPadringPinAttrStrongestDrive = 1 << 8,
+  kDifPinmuxPadringPinAttrSchmittTrigger = 1 << 9,
+  kDifPinmuxPadringPinAttrSlewRateSlow = 1 << 10,
+  kDifPinmuxPadringPinAttrSlewRateFast = 1 << 11,
+} dif_pinmux_padring_pin_attr_t;
+
+/**
+ * A sleep mode for a DIO or MIO pin.
+ *
+ * This determines the behaviour of a pin in the deep sleep state.
+ *
+ * Note:
+ * After the chip has come back from the deep sleep, the mode of the pin
+ * persists until is explicitly cleared by `dif_pinmux_sleep_clear_state`.
```

Comment:
nit: ...until is... -> ...until **it** is...

Reply:
Done.